app/models/invoice.rb

```diff
@@ -8,4 +8,20 @@ class Invoice < ApplicationRecord
   has_many :transactions
   has_many :invoice_items
   has_many :items, through: :invoice_items
+
+  def find_invoice_item(invoice, item)
+    InvoiceItem.where(invoice: invoice, item: item)
+  end
+
+  def revenue_order
+    items.
+    joins(:invoice_items).
+    select('items.*, sum(invoice_items.quantity *invoice_items.unit_price) as revenue').
+    group('items.id').
+    order(revenue: :desc)
+  end
+
+  def total_revenue
+    revenue_order.collect {|item| item.revenue }.sum
```

Comment:
Could you refactor this using what we learned in class today with group and sum aggregates so we are just using activerecord?

Reply:
Okay, I refactored this method to look less ruby-ish!